examples/software-device/rs-software-device.cpp

```diff
@@ -0,0 +1,185 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2017 Intel Corporation. All Rights Reserved.
+
+#include <librealsense2/rs.hpp> // Include RealSense Cross Platform API
+#include <librealsense2/hpp/rs_internal.hpp>
+
+#include "example.hpp" 
+#include "example.hpp"          // Include short list of convenience functions for rendering
+
+
+#define STB_IMAGE_WRITE_IMPLEMENTATION
+#include <../third-party/stb_image_write.h>
+#include <../common/res/int-rs-splash.hpp>
+
+#define STB_IMAGE_IMPLEMENTATION
+#include <../third-party/stb_image.h>
+
+using namespace rs2;
+
+struct synthetic_frame
+{
+    int x, y, bpp;
+    std::vector<uint8_t> frame;
+};
+
+synthetic_frame create_synthetic_texture()
+{
+    synthetic_frame frame;
+    auto r = stbi_load_from_memory(splash, (int)splash_size, &frame.x, &frame.y, &frame.bpp, false);
```

Comment:
`auto r` is the least expressive way to go here...please consider using the actual type and providing a proper name for `r`.